pkg/probes/curl/curl_json.go

```diff
@@ -204,6 +206,15 @@ func (clp Probe) ParseProbeOutput(probeOutput string, outputDestination *output.
 				[]string{fmt.Sprintf("%s (%s)", url, probeResult.ErrorMsg)},
 			)
 		}
+		if ensurePrivate {
```

Comment:
Could we add some comments here and in this function's docstring explaining the usecase for `ensurePrivate`?